### ISSUE TITLE
Not cop checks in string literals 

### ISSUE THREAD

**Issue by snowe2010** [bbatsov/rubocop#1874]
`visible = shld.eql? 'not ' ? false : true`

gives the warning:

`W: Literal 'not ' appeared in a condition.`

**Comment by snowe2010** [comment]
Ah never mind, I see that it's actually finding a literal in the statement. Why isn't this allowed and what is the alternative solution?

**Comment by bbatsov** [comment]
A literal is always true or false, so the condition makes no sense.

**Comment by bquorning** [comment]
I had to go look up the precedence rules, but you’re correct. Ruby interprets the example as
```rb
visible = shld.eql?('not ' ? false : true)
```
which always evaluates to 
```rb
visible = false
```
